ir/instr.cpp

```diff
@@ -1911,4 +1911,125 @@ unique_ptr<Instr> Memcpy::dup(const string &suffix) const {
   return make_unique<Memcpy>(*dst, *src, *bytes, align_dst, align_src, move);
 }
 
+
+vector<Value*> ExtractElement::operands() const {
+  return { v, idx };
+}
+
+void ExtractElement::rauw(const Value &what, Value &with) {
+  RAUW(v);
+  RAUW(idx);
+}
+
+void ExtractElement::print(ostream &os) const {
+  os << getName() << " = extractelement " << *v << ", " << *idx;
+}
+
+StateValue ExtractElement::toSMT(State &s) const {
+  auto &[iv, ip] = s[*idx];
+  auto vty = v->getType().getAsAggregateType();
+  expr inbounds = !iv.uge(vty->numElementsConst());
```

Comment:
you can use ult here.

Reply:
> Please use ule/uge whenever possible (cheaper in Z3)

Hi @nunoplopes , On the last round of review, you mentioned that use "ule/uge" whenever possible. Does the performance difference between "equal" and "than" get fixed or am I missed something?

Reply:
Right, you should use uge/ule when it's possible. I mean, when it's possible to change the formula around such that you don't need to do !uge (which is the same as ult). When you really need to do ult, just use it.

Reply:
fixed :) Thanks for the explanation!!!